enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,451 @@
+[providers](https://github.com/kubernetes-sigs/external-dns/tree/master/provider) (at least, initially).
+
+## Proposal
+
+The proposal is based on experience gathered and work accomplished with existing OpenShift operators.
+
+The following tasks would be completed as a part of this enhancement:
+
+* Create a GitHub repository to host the operator source code in OpenShift's Github org,
+with the hopes of moving the operator's source code into an upstream k8s-sigs repository in the future.
+* Leverage popular frameworks and libraries , i.e. controller-runtime and kubebuilder, to simplify development
+of the operator and align with upstream communities.
+* Manage dependencies through [Go Modules](https://blog.golang.org/using-go-modules).
+* Create user and developer documentation for running ExternalDNS operator (on both OpenShift and vanilla Kubernetes).
+* Create tooling to simplify building, running, testing, etc. the operator.
+* Create tests that reduce bugs in the code and allow for continuous integration (tests should be compatible
+with OpenShift CI, in addition to vanilla kubernetes environments running in public clouds).
+* Integrate the operator with the OpenShift toolchain, i.e. [openshift/release](https://github.com/openshift/release),
+which would be required for productization.
+* Provide manifests for quickly deploying the operator (including a minimal ExternalDNS Operator CR manifest).
+* Create tooling to simplify ExternalDNS troubleshooting. For example, add support for the Operator to
+OpenShift [must-gather](https://github.com/openshift/must-gather).
+* Integrate ExternalDNS with the OpenShift monitoring toolchain. For example, add
+[metrics support](https://github.com/openshift/cluster-monitoring-operator) for the operator and its operand(s).
+
+
+In addition, the following functionality is expected as part of the operator (note that this is not an exhaustive list):
+
+* Introduce a CRD to be used by the operator to manage ExternalDNS and its dependencies.
+* Common fields in spec that define the configuration of ExternalDNS and any dependent resources,
+i.e. a secret resource containing credentials for the necessary cloud provider.
+* Common fields in spec that define how to deploy ExternalDNS.
+* Common fields in status that expose the current health & version information of ExternalDNS operands.
+* The operator will be capable of observing other resources to perform basic sequencing required for OpenShift
+integration.
+* The operator will be capable of deploying multiple instances of ExternalDNS (operands), which is necessary when
+a common domain is shared across DNS zones, or when using multiple DNS Providers. See ExternalDNS' upstream
+[AWS public/private zone tutorial docs](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/public-private-route53.md)
+as an example.
+
+Here is what a CRD instance for ExternalDNS Operator may look like:
+
+```yaml
+apiVersion: operator.externaldns.io/v1alpha1
```

Comment:
for the externaldns.io domain, is there a corresponding api compatibility level or guidance we need to account for going forward.

see: https://github.com/openshift/openshift-docs/pull/30286

Reply:
Looks like upstream externalDNS actually uses `externaldns.k8s.io` (https://github.com/kubernetes-sigs/external-dns/blob/master/docs/contributing/crd-source/crd-manifest.yaml).

I'll update this section accordingly.